### DISCUSSION TITLE
Filter status badges by build label

### DISCUSSION THREAD

**Discussion by daniel-naegele** [codemagic-ci-cd/codemagic-docs#1918] [Ideas]
### Use case

As per the [White Label apps](https://docs.codemagic.io/yaml-quick-start/white-label-getting-started/) instructions, we use the same workflow with different variables and different labels.

Thus we want to have a table with all the latest builds for the different clients. Unfortunately we can't do that, as there is no way to differentiate between labeled builds in the badge link as far as I know.

### Idea

Something like an optional GET parameter could be used, like
`https://api.codemagic.io/apps/<app-id>/<workflow-id>/status_badge.svg?label=<label>`

In addition, but this is not really necessary, this could also be implemented in the `https://codemagic.io/apps/<app-id>/<workflow-id>/latest_build`url in order to automatically get to the latest build with the right label.

**Comment by Ycoi** [comment]
Thanks for the idea!

Did I get this right, you would wnat the API to filter to return the status only for specific label? And may I ask what the use case is here? Are you using Codemagic and the labels for whitelabeling and thus use a single workflow with different labels, or do you have some other use case.

**Reply by daniel-naegele** [reply]
Yes, this is exactly our use case. We use the same workflows (android/ios split up) for all clients and attach a label with the client id to the builds, just as the documentation suggests. 
With filtering the badges for the same workflow with different labels, we could easily maintain a build matrix to see, which clients had a successful deploy and which not.

**Reply by Ycoi** [reply]
Thanks for your reply. I discussed this further with our dev team and the badge query is a bit special case and we don’t have current plans on adding this functionality. 
I think a more generic solution would be to make it possible to filter the returned builds of the Codemagic Builds API with labels. We are planning on implementing a new ipmroved API on the future so I marked this down as one thing we could add. 

But I think this is something you could do already now on your side, fetch builds with the builds API and then check for the build labels and statuses on your side. 

Also, is there some reason why you can't use [Codemagic Build Dashboards ](https://docs.codemagic.io/yaml-publishing/build-dashboards/) for this? That’s a quite convenient way to make dashboards of your builds and you can easily filter those with status and labels etc.

**Reply by daniel-naegele** [reply]
What I want to achieve in the end, is to have a section in our README with a table containing our clients/flavors as rows and build status badges for iOS/Android being each one column. With the proposed build badges I could just add the badges and could see at a glance which builds for which clients/flavors for which platforms have failed and which not, which are currently running/scheduled, have never run ... 

I can't achieve this (for me easy to implement) with builds, as they require extra processing logic and authenticated access to the codemagic API. Codemagic Build Dashboards are way too tedious, imagine having more than 20 flavors, setting each one up, clicking on each of one to check build statuses. The solution with the README section is way easier and more helpful (at least for our use case).

**Comment by Ycoi** [comment]
I will close this thread since we unfortunately don’t have plans to add this feature in the foreseeable future, at least not for the build badges since there are some extra complications introduced by how the build badges work. We might extend the API at some point to add the filtering we talked about above, but also that’s not in current planning.

Thanks a lot for the idea though, and if you have some other ideas or feature requests, please open a new thread for those.
